usl_pipeline/study_area_uploader/study_area_uploader/transformers/soil_classes_transformers.py

```diff
@@ -22,32 +24,54 @@ def transform_soil_classes_as_green_areas(
         elevation_header: The source of coordinate system and study area region info.
         green_areas_polygons: List of tuples with green area polygons.
         soil_classes_polygons: List of tuples with soil class polygons.
+        log_details: Indicates that details of intermediate steps should be logged.
 
     Returns:
         Corrected soil classes that can be used as green areas.
     """
+    if log_details:
```

Comment:
Rather than have branches for logging statements, what we can do is this:
```python
def main():
  args = parser.parse_args()
  if args.verbose:
    logging.basicConfig(level=logging.INFO)
  else:
    logging.basicConfig(level=logging.ERROR)
    
def foo():
  logging.info('whatever')
  logging.error('oh no!')
```
so the `logging.basicConfig(level=LEVEL)` call determines what level of logging calls are actually emitted and each function can just make `logging.level()` calls without needing to pass `log_details` arguments everywhere.

Reply:
Yeah, that's what I was looking for! Done.